modules/debts_api/lib/debts_api/v0/fsr_form_transform/enhanced_expense_calculator.rb

```diff
@@ -0,0 +1,113 @@
+            exclude_by(@expense_records, [RENT, MORTGAGE_PAYMENT])
+          )
+          @all_expenses ||= get_all_expenses
+        end
+
+        def get_monthly_expenses
+          utilities = get_utilities
+          installments = safe_sum(@installment_contracts.pluck('amountDueMonthly'))
+          other_exp = safe_sum(@other_expenses.pluck('amount'))
+          calculated_exp_records = safe_sum(@expense_records.pluck('amount'))
+          food = safe_number(@old_food_attr)
+          rent_or_mortgage_sum = safe_number(@old_rent_mortgage_attr)
+          credit_card_bills = safe_sum(@credit_card_bills.pluck('amountDueMonthly'))
+
+          sum = utilities +
+                installments +
+                other_exp +
+                calculated_exp_records +
+                food +
+                rent_or_mortgage_sum +
+                credit_card_bills
+
+          sum.round(2)
+        end
+
+        def get_all_expenses
+          {
+            rentOrMortgage: get_rent_mortgage_expenses,
+            food: get_food_expenses,
+            utilities: get_utilities,
+            otherLivingExpenses: get_other_living_expenses,
+            expensesInstallmentContractsAndOtherDebts: get_installments_and_other_debts,
+            otherExpenses: @other_expenses,
+            filteredExpenses: @filtered_expenses,
+            installmentContractsAndCreditCards: [@installment_contracts, @credit_card_bills].flatten
+          }
+        end
+
+        private
+
+        def get_rent_mortgage_expenses
+          safe_number(@old_rent_mortgage_attr)
+          rent_or_mortgage_expenses = @expense_records.filter do |record|
+            [RENT, MORTGAGE_PAYMENT].include?(record['name'])
+          end
+          safe_sum(rent_or_mortgage_expenses.pluck('amount'))
+        end
+
+        def get_food_expenses
+          food_expenses = @other_expenses.find { |expense| expense['name'] == 'Food' } || { amount: 0 }
+          safe_number(food_expenses['amount'])
+        end
+
+        def get_utilities
+          amounts = @utility_records.pluck('amount')
+          safe_sum(amounts)
+        end
+
+        def get_other_living_expenses
+          {
+            name: @filtered_expenses.pluck('name').join(', '),
+            amount: safe_sum(@filtered_expenses.pluck('amount'))
+          }
+        end
+
+        def get_installments_and_other_debts
+          installment_monthly_due = @installment_contracts.pluck('amountDueMonthly')
+          credit_card_monthly_due = @credit_card_bills.pluck('amountDueMonthly')
+          safe_sum([installment_monthly_due, credit_card_monthly_due].flatten)
+        end
+
+        def exclude_by(expenses, names)
+          expenses.filter { |expense| names.exclude?(expense['name']) }
+        end
+
+        def safe_number(str)
+          return 0.0 if str.nil?
+
+          str.gsub(/[^0-9.-]/, '').to_f
+        end
+
+        def safe_sum(ary)
+          ary.map { |el| safe_number(el) }.sum.round(2)
```

Comment:
I don't know the precision required but it is safer to calculate using integers (cents) than floating point numbers.